app/scripts/keyring-controller.js

```diff
@@ -173,10 +173,15 @@ module.exports = class KeyringController extends EventEmitter {
     })
   }
 
-  placeSeedWords () {
+  placeSeedWords (cb) {
     const firstKeyring = this.keyrings[0]
     const seedWords = firstKeyring.serialize().mnemonic
     this.configManager.setSeedWords(seedWords)
+
+    if (cb) {
+      cb()
+      this.emit('update')
```

Comment:
We only call it synchronously in `createNewVaultAndKeychain()`, which manages its own updates, which is why it needs to be called in the `cb` version and not in the sync version, but since that's hard to follow, I'll just move it.